interface/wx/uilocale.h

```diff
@@ -218,15 +218,36 @@ class wxLocaleIdent
 
     /**
         Set script.
+
+        Note that script value is currently ignored under Unix systems.
+
         Return reference to @this for method chaining.
 
         @param script
             It is an initial-uppercase ISO 15924 script code.
     */
     wxLocaleIdent& Script(const wxString& script);
 
+    /**
+        Set charset.
+
+        Note that this value is only used under Unix systems and simply ignored
+        under the other ones.
+
+        Return reference to @this for method chaining.
+
+        @param charset
+            Charset is a string such as "UTF-8", "ISO855915" or "KOI8R".
+            Supported charsets depend on the implementation and installation.
+    */
+    wxLocaleIdent& Charset(const wxString& charset);
```

Comment:
I don't think this should exist here, especially if it's for use with wxUILocale — any support *UI* toolkit operates on Unicode and (on Unix) UTF-8. That locale on some Unices includes charset is a legacy thing that makes little sense in a new API. Can't the utf8 think mentioned in the comment be tried transparently when setting the locale?

Reply:
I was going to not have it initially, but then decided that we were going to need it to support parsing Unix locale strings and added it. And I think encodings such as KOI8 or Shift-JIS might still be used in some places, although I admit I am not really sure.

And we do already try UTF-8 (just as we did with `wxLocale`) under Unix, but we actually use this method in the implementation, so removing it would need changing that code and making it less nice (right now all functions take just `wxLocaleIdent`), so I'd rather keep it, unless you really, really hate it. But I'll update the comment to make it more dissuasive.